packages/api-tests/test/window.spec.js

```diff
@@ -0,0 +1,92 @@
+const assert = require('assert');
+
+module.exports = (setup, mocha) => {
+  const {
+    beforeEach,
+    afterEach,
+    describe,
+    it
+  } = mocha;
+
+  let app;
+
+  describe('Window API', function(done) {
+    const timeout = 60000;
+    this.timeout(timeout);
+
+    beforeEach(() => {
+      app = setup(timeout);
+
+      return app.start();
+    });
+
+    afterEach(function() {
+      if (app && app.isRunning()) {
+        return app.stop();
+      }
+    });
+
+    const executeAsyncJavascript = (client, script) => {
+      return client.executeAsync(script);
+    };
+
+    it('Check ssf.Window is available globally', function() {
+      const script = `
+        var callback = arguments[arguments.length - 1];
+        if (ssf.Window !== undefined) {
+          callback();
+        }
+      `;
+      return executeAsyncJavascript(app.client, script);
+    });
+
+    it('Check window constructor opens a new window', function() {
+      const script = `
+        ssf.app.ready().then(() => {
+          var callback = arguments[arguments.length - 1];
+          new ssf.Window({url: 'about:blank', name: 'test', show: true, child: true});
+          setTimeout(() => callback(), 500);
+        });
+      `;
+      return executeAsyncJavascript(app.client, script).then((result) => {
+        return app.client.getWindowCount().then((count) => {
+          assert.equal(count, 2);
+        });
+      });
+    });
+
+    it('Check new window has correct x position', function() {
```

Comment:
I think for tests involving multiple windows, yes. This is because we probably want to run JavaScript in window 1, then switch to window 2, then run JavaScript in window 2, then check for some value/action. Maybe we could extract most of the JavaScript string that gets passed in some way? Something like:
```
const windowMethod = (method) => {
   exectueJavaScript(client, '
     var callback = ...
     ssf.Window.getCurrent().${method}().then(() => callback())
  ')
}
```
Which would at least reduce a lot of the boiler plate in the tests somewhat?